sky/skylet/providers/kubernetes/node_provider.py

```diff
@@ -61,24 +61,28 @@ def __init__(self, provider_config, cluster_name):
         # kubernetes setups.
         self.timeout = provider_config['timeout']
 
-    def non_terminated_nodes(self, tag_filters):
+    def non_terminated_nodes(self, tag_filters, override_all_nodes=False):
```

Comment:
I tried to use the same method of if `override_all_nodes` is true, then it returns all pods, otherwise it constructs `field_selector` to filter returned pods.